packages/core/src/sync-realtime/filter.ts

```diff
@@ -12,7 +7,31 @@ import {
 } from "@/sync/source.js";
 import type { SyncBlock, SyncCallTrace, SyncLog } from "@/types/sync.js";
 import { toLowerCase } from "@/utils/lowercase.js";
-import { hexToNumber } from "viem";
+import { type Address, hexToNumber } from "viem";
+
+const isValueMatched = <T extends string>(
+  filterValue: T | T[] | null | undefined,
+  eventValue: T | undefined,
+): boolean => {
+  // match all
+  if (filterValue === null || filterValue === undefined) return true;
+
+  // missing value
+  if (eventValue === undefined) return false;
+
+  // array
+  if (
+    Array.isArray(filterValue) &&
+    filterValue.some((v) => v === toLowerCase(eventValue))
```

Comment:
Is `filterValue` already expected to be lowercase here?

Reply:
yes